### PR TITLE
Use req.VerifyWithCB to search for missing DARCs

### PR DESCRIPTION
**Pull request by talhaparacha**
@ineiti suggested this fix to get the Omniledger service working.

### CODE REVIEW

File: omniledger/service/service.go [hunk 1]
```diff
@@ -286,8 +287,18 @@ func (s *Service) verifyInstruction(scID skipchain.SkipBlockID, instr Instructio
 	if err != nil {
 		return errors.New("couldn't create darc request: " + err.Error())
 	}
-	// TODO we need to use req.VerifyWithCB to search for missing darcs
-	err = req.Verify(d)
+	err = req.VerifyWithCB(d, func(str string, latest bool) *darc.Darc {
+		log.Printf("-%s-", str)
```
Comment (ineiti) [inline omniledger/service/service.go]: `log.Print` are for debugging, please remove them for inclusion in master.

File: omniledger/service/service.go [hunk 2]
```diff
@@ -286,8 +287,18 @@ func (s *Service) verifyInstruction(scID skipchain.SkipBlockID, instr Instructio
 	if err != nil {
 		return errors.New("couldn't create darc request: " + err.Error())
 	}
-	// TODO we need to use req.VerifyWithCB to search for missing darcs
-	err = req.Verify(d)
+	err = req.VerifyWithCB(d, func(str string, latest bool) *darc.Darc {
```
Comment (ineiti) [inline omniledger/service/service.go]: Add a short comment to say what this will be doing.

**Comment by talhaparacha** [comment]
@ineiti Please review now.

**Comment by ineiti** [comment]
OK - looks good. What I forgot is to have a test for it... As you already know how to write darcs to omniledger, can you copy `TestService_DarcEvolution` to `TestService_DarcDelegate` and create and signature with a delegated darc? That would round up nicely the addition!

**Pull request merged**